lib/internal/Magento/Framework/MultiDimensionalIndexer/IndexHandlerInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\MultiDimensionalIndexer;
+
+/**
+ * Represents manipulation with index data
+ *
+ * @api
+ */
+interface IndexHandlerInterface
+{
+    /**
+     * Add data to index
+     *
+     * @param IndexName $indexName
+     * @param \Traversable $documents
+     * @param string $connectionName
+     * @return void
+     */
+    public function saveIndex(IndexName $indexName, \Traversable $documents, string $connectionName);
```

Comment:
What the purpose of using connectionName as an argument for index handler? 
Why this is not a responsibility of index class?